packages/web-media-box/src/hls-parser/utils/parse.ts

```diff
@@ -19,3 +21,19 @@ export const parseHex = (val: string): ArrayBuffer | undefined => {
 
   return new Uint8Array(hexes as unknown as ArrayLike<number>).buffer as ArrayBuffer;
 };
+
+export const parseEncryptionTag = (tagAttributes: Record<string, string>): Encryption | SessionKey => {
+  const METHOD = 'METHOD';
+  const URI = 'URI';
+  const IV = 'IV';
+  const KEYFORMAT = 'KEYFORMAT';
+  const KEYFORMATVERSIONS = 'KEYFORMATVERSIONS';
+
+  return {
+    method: tagAttributes[METHOD] as 'NONE' | 'AES-128' | 'SAMPLE-AES',
+    uri: tagAttributes[URI],
+    iv: tagAttributes[IV],
+    keyFormat: tagAttributes[KEYFORMAT] || 'identity',
+    keyFormatVersions: tagAttributes[KEYFORMATVERSIONS] ? tagAttributes[KEYFORMATVERSIONS].split('/').map(Number) : [1],
+  };
+};
```

Comment:
If you want to re-use static fields instead of re-define these constants here

we can do the following:

```ts
abstract class EncryptionTagProcessor extends TagWithAttrbibutesProcessor {
  parseEncryption() { ... }
}

class ExtXKey extends EncryptionTagProcessor {
...
}

class ExtXSessionKey extends EncryptionTagProcessor {
...
}

```

Reply:
Ahh yes, proper OOP. 😁  Thanks for the remind, it's been a while. Fix incoming.

Reply:
fixed